labgrid/util/agents/sysfsgpio.py

```diff
@@ -0,0 +1,68 @@
+"""
+This module implements implements switching GPIOs via sysfs GPIO kernel interface.
+
+Takes an integer property 'absolute_index' which refers to the already exported GPIO device.
+
+"""
+import logging
+import os
+
+class GpioDigitalOutput:
+    _gpio_sysfs_path_prefix = '/sys/class/gpio'
+
+    def __init__(self, **kwargs):
+        index = kwargs['index']
+        self._logger = logging.getLogger("Device: ")
+        gpio_sysfs_path = os.path.join(GpioDigitalOutput._gpio_sysfs_path_prefix, 'gpio{0}'.format(index))
+        if not os.path.exists(gpio_sysfs_path):
+            raise ValueError("Device not found")
+        gpio_sysfs_direction_path = os.path.join(gpio_sysfs_path, 'direction')
+        direction_fd = os.open(gpio_sysfs_direction_path, flags=os.O_WRONLY)
+        self._logger.debug("Configuring GPIO {idx} as output.".format(idx = index))
+        os.write(direction_fd, b'out')
+        os.close(direction_fd)
+        gpio_sysfs_value_path = os.path.join(gpio_sysfs_path, 'value')
+        self.gpio_sysfs_value_fd = os.open(gpio_sysfs_value_path, flags=(os.O_RDWR | os.O_SYNC))
+
+    def __del__(self):
+        os.close(self.gpio_sysfs_value_fd)
+        self.gpio_sysfs_value_fd = None
+
+    def get(self):
+        os.lseek(self.gpio_sysfs_value_fd, 0, os.SEEK_SET)
+        literal_value = os.read(self.gpio_sysfs_value_fd, 1)
+        if literal_value == b'0':
+            return False
+        elif literal_value == b'1':
+            return True
+        raise ValueError("GPIO value is out of range.")
+
+    def set(self, status):
+        self._logger.debug(
+                "Setting GPIO to `{}`.".format(status))
+        binary_value = None
+        if status is True:
+            binary_value = b'1'
+        elif status is False:
+            binary_value = b'0'
+
+        if binary_value is None:
+            raise ValueError("GPIO value is out of range.")
+
+        os.write(self.gpio_sysfs_value_fd, binary_value)
+
+
+def handle_set(_index, status):
+    gpio_line = GpioDigitalOutput(index=_index)
```

Comment:
You should cache the instances in a module level dictionary. That would avoid reconfiguring the GPIO for every call.